lib/remoteproc/remoteproc_virtio.c

```diff
@@ -16,6 +16,89 @@
 #include <metal/utilities.h>
 #include <metal/alloc.h>
 
+static void rproc_virtio_delete_virtqueues(struct virtio_device *vdev)
+{
+	struct virtio_vring_info *vring_info;
+	unsigned int i;
+
+	if (!vdev->vrings_info)
+		return;
+
+	for (i = 0; i < vdev->vrings_num; i++) {
+		vring_info = &vdev->vrings_info[i];
+		if (vring_info->vq)
+			virtqueue_free(vring_info->vq);
+	}
+}
+
+static int rproc_virtio_create_virtqueue(struct virtio_device *vdev,
+					 unsigned int flags,
+					 unsigned int idx,
+					 const char *name,
+					 vq_callback callback)
+{
+	struct virtio_vring_info *vring_info;
+	struct vring_alloc_info *vring_alloc;
+	int ret;
+	(void)flags;
+
+	/* Get the vring information */
+	vring_info = &vdev->vrings_info[idx];
+	vring_alloc = &vring_info->info;
+
+	/* Virtqueue shouldn't have been created */
```

Comment:
Minor: This comment is misleading. The code would be clearer if it was left out. A more accurate comment would be something like:
"Fail if the virtqueue has already been created"

Reply:
Addressed, thanks, looks better now.